File: src/main/java/com/ptaf/ai/validation/StepReuseValidationResult.java
package com.ptaf.ai.validation;

import java.util.List;

public record StepReuseValidationResult(
        List<String> featureSteps,
        List<String> matchedExistingSteps,
        List<String> unmatchedSteps,
        List<String> claimedReusedButNotFound,
        List<String> claimedNewButAlreadyExists,
        int totalSteps,
        int matchedCount,
        int unmatchedCount,
        double reusePercentage,
        boolean passed,
        List<String> warnings
) {
}

File: src/main/ptaf/ai/validation/step_reuse_validation_result.py
from __future__ import annotations

from dataclasses import dataclass


@dataclass(frozen=True)
class StepReuseValidationResult:
    feature_steps: list[str]
    matched_existing_steps: list[str]
    unmatched_steps: list[str]
    claimed_reused_but_not_found: list[str]
    claimed_new_but_already_exists: list[str]
    total_steps: int
    matched_count: int
    unmatched_count: int
    reuse_percentage: float
    passed: bool
    warnings: list[str]
